products/llm_observability/frontend/LLMObservabilityTraceScene.tsx

```diff
@@ -180,9 +182,25 @@ function TraceSidebar({
     tree: EnrichedTraceTreeNode[]
 }): JSX.Element {
     const ref = useRef<HTMLDivElement | null>(null)
-    const { searchQuery, mostRelevantEvent } = useValues(llmObservabilityTraceDataLogic)
+    const { mostRelevantEvent, searchOccurrences } = useValues(llmObservabilityTraceDataLogic)
+    const { searchQuery } = useValues(llmObservabilityTraceLogic)
     const { setSearchQuery, setEventId } = useActions(llmObservabilityTraceLogic)
 
+    const [searchValue, setSearchValue] = useState(searchQuery)
```

Comment:
I know I used a use effect below, but this use case should be in Kea

Reply:
This is really local to this component though, it's to show the new character typed in the search bar, whereas the `setSearchQuery` is a kea action and it's debounced to only initiate the search after a pause.

Do you feel strongly about this living in kea?